src/servers/http_v2_server.cc

```diff
@@ -0,0 +1,877 @@
+          TIMESPEC_TO_NANOS(req->recv_end_ts));
+    }
+  }
+#endif  // TRTIS_ENABLE_TRACING
+
+  std::string infer_request_header(
+      evhtp_kv_find(req->headers_in, kInferRequestHTTPHeader));
+
+  InferRequestHeader request_header;
+  if (!google::protobuf::TextFormat::ParseFromString(
+          infer_request_header, &request_header)) {
+    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    return;
+  }
+
+  std::string request_header_serialized;
+  if (!request_header.SerializeToString(&request_header_serialized)) {
+    evhtp_send_reply(req, EVHTP_RES_BADREQ);
+    return;
+  }
+
+  uint64_t unique_id = RequestStatusUtil::NextUniqueRequestId();
+
+  // Create the inference request provider which provides all the
+  // input information needed for an inference.
+  TRTSERVER_InferenceRequestProvider* request_provider = nullptr;
+  TRTSERVER_Error* err = TRTSERVER_InferenceRequestProviderNew(
+      &request_provider, server_.get(), model_name.c_str(), model_version,
+      request_header_serialized.c_str(), request_header_serialized.size());
+  if (err == nullptr) {
+    EVBufferPair* response_pair(new EVBufferPair());
+    err = EVBufferToInput(
+        model_name, request_header, req->buffer_in, request_provider,
+        response_pair->second);
+    if (err == nullptr) {
+      InferRequestClass* infer_request = new InferRequestClass(
+          req, request_header.id(), server_id_, unique_id);
+
+      response_pair->first = req->buffer_out;
+      infer_request->response_pair_.reset(response_pair);
+
+      // Get the trace object to use for this request. If nullptr then
+      // no tracing will be performed.
+      TRTSERVER_Trace* trace = nullptr;
+#ifdef TRTIS_ENABLE_TRACING
+      if (tracer != nullptr) {
+        infer_request->tracer_ = std::move(tracer);
+        trace = infer_request->tracer_->ServerTrace();
+      }
+#endif  // TRTIS_ENABLE_TRACING
+
+      err = TRTSERVER_ServerInferAsync(
+          server_.get(), trace, request_provider, allocator_,
+          reinterpret_cast<void*>(response_pair),
+          InferRequestClass::InferComplete,
+          reinterpret_cast<void*>(infer_request));
+      if (err != nullptr) {
+        delete infer_request;
+        infer_request = nullptr;
+      }
+    }
+  }
+
+  // The request provider can be deleted before ServerInferAsync
+  // callback completes.
+  TRTSERVER_InferenceRequestProviderDelete(request_provider);
+
+  if (err != nullptr) {
+    RequestStatus request_status;
+    RequestStatusUtil::Create(&request_status, err, unique_id, server_id_);
+
+    InferResponseHeader response_header;
+    response_header.set_id(request_header.id());
+    evhtp_headers_add_header(
+        req->headers_out,
+        evhtp_header_new(
+            kInferResponseHTTPHeader,
+            response_header.ShortDebugString().c_str(), 1, 1));
+    LOG_VERBOSE(1) << "Infer failed: " << request_status.msg();
+
+    evhtp_headers_add_header(
+        req->headers_out, evhtp_header_new(
```

Comment:
Did you run the formatter... this looks strange.

Reply:
I hadn't resolved now